checkov/terraform/checks/resource/aws/ECRPolicy.py

```diff
@@ -21,9 +19,16 @@ def scan_resource_conf(self, conf):
         :param conf: aws_ecr_repository configuration
         :return: <CheckResult>
         """
-        if "policy" in conf.keys() and is_json(conf["policy"][0]) \
-                and json.loads(conf["policy"][0])['Statement'][0]['Principal'] == '*':
-            return CheckResult.FAILED
+        if "policy" in conf.keys():
+            policy = conf["policy"][0]
+            if policy == '':
+                return CheckResult.PASSED
+            if policy['Statement'][0]:
+                statement = policy['Statement'][0]
+                if statement['Principal']:
+                    principal = statement['Principal']
+                    if principal == "*":
+                        return CheckResult.FAILED
```

Comment:
@JamesWoolfenden can you add some type checks? That policy is a list, that policy[0] is a dict etc